<launch>
    <param name="/use_sim_time" value="false" />

    <group>
        <remap from="tf" to="tf_old" />
        <node name="broadcaster_ab" pkg="tf" type="static_transform_publisher" args="1 2 3 4 5 6 a b 10"/>
        <node name="broadcaster_de" pkg="tf" type="static_transform_publisher" args="0 0 0 0 0 0 d e 10"/>
        <node name="broadcaster_fg" pkg="tf" type="static_transform_publisher" args="0 0 0 0 0 0 f g 10"/>
        <node name="broadcaster_hi" pkg="tf" type="static_transform_publisher" args="0 0 0 0 0 0 h i 10"/>
        <node name="broadcaster_ij" pkg="tf" type="static_transform_publisher" args="0 0 0 0 0 0 i j 10"/>
        <node name="broadcaster_kl" pkg="tf" type="static_transform_publisher" args="0 0 0 0 0 0 k l 10"/>

        <node name="broadcaster_yz" pkg="tf" type="static_transform_publisher" args="0 0 0 0 0 0 y z 10"/>
    </group>

    <node name="remapper" pkg="tf_remapper_cpp" type="tf_remap" output="screen">
        <rosparam param="mappings">[{old: b, new: c}, {old: d, new: ''}, {old: '', new: d}, {old: g}, {old: i}, {old: k, new: k}]</rosparam>
        <param name="is_bidirectional" value="true" />
    </node>

    <test test-name="tf_remapper_node" pkg="tf_remapper_cpp" type="test_tf_remapper_node" />
</launch>